grades/signals.py

```diff
@@ -20,6 +20,16 @@ def handle_create_coursecertificate(sender, instance, created, **kwargs):  # pyl
         transaction.on_commit(lambda: generate_program_certificate(user, program))
 
 
+@receiver(post_save, sender=MicromastersProgramCertificate, dispatch_uid="programcertificate_post_save")
+def handle_create_programcertificate(sender, instance, created, **kwargs):  # pylint: disable=unused-argument
+    """
+    When a MicromastersProgramCertificate model is created
+    """
+    if created:
+        user = instance.user
+        transaction.on_commit(lambda: generate_program_letter(user, instance.course.program))
```

Comment:
I tried to enroll a user in all courses of a program. Then I tried to create a certificate for each course. Got an error in the signals on the last certificate creation.
```
AttributeError: 'MicromastersProgramCertificate' object has no attribute 'course'
```
Let me know if I missed anything here.